test/e2e/runner/load.go

```diff
@@ -40,31 +41,39 @@ func Load(ctx context.Context, testnet *e2e.Testnet) error {
 		}
 
 		for w := 0; w < testnet.LoadTxConnections; w++ {
-			go loadProcess(ctx, txCh, chSuccess, n)
+			go loadProcess(ctx, txCh, chSuccess, chFailed, n)
 		}
 	}
 
-	// Monitor successful transactions, and abort on stalls.
-	success := 0
+	// Monitor successful and failed transactions, and abort on stalls.
+	success, failed := 0, 0
 	timeout := initialTimeout
 	for {
+		rate := log.NewLazySprintf("%.1f", float64(success)/time.Since(started).Seconds())
+
 		select {
 		case <-chSuccess:
 			success++
-			if testnet.LoadMaxTxs > 0 && success >= testnet.LoadMaxTxs {
-				logger.Info("load", "msg", log.NewLazySprintf("Ending transaction load after reaching %v txs (%.1f tx/s)...",
-					success, float64(success)/time.Since(started).Seconds()))
-				return nil
-			}
 			timeout = stallTimeout
+		case <-chFailed:
+			failed++
 		case <-time.After(timeout):
 			return fmt.Errorf("unable to submit transactions for %v", timeout)
 		case <-ctx.Done():
 			if success == 0 {
 				return errors.New("failed to submit any transactions")
 			}
-			logger.Info("load", "msg", log.NewLazySprintf("Ending transaction load after %v txs (%.1f tx/s)...",
-				success, float64(success)/time.Since(started).Seconds()))
+			logger.Info("load", "msg", log.NewLazySprintf("Ending transaction load after %v txs (%.1f tx/s)...", success, rate))
+			return nil
+		}
+
+		// Log every ~1 second the number of sent transactions.
+		if (success+failed)%testnet.LoadTxBatchSize == 0 {
```

Comment:
here you added `success+failed` which validates the comment above about the rate

Reply:
Here we actually want to produce a log per second, and this works because `success+failed` is equal to `LoadTxBatchSize * time.Since(started).Seconds()`, so `(success+failed)%testnet.LoadTxBatchSize` will be 0 every time a batch of txs is sent (successfully or not).